app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -203,7 +206,16 @@ class HomeFragment : Fragment(), CoroutineScope {
                 (activity as HomeActivity).openToBrowser(BrowserDirection.FromHome)
             }
             is TabAction.Close -> {
-                removeTabWithUndo(action.sessionId)
+                if (deleteSessionJob == null) removeTabWithUndo(action.sessionId) else {
```

Comment:
This only buffers a single job. A user could swipe away an indeterminate number of tabs while the snackbar is active or one could move to another screen. I believe the best long-term solution if we want undo capabilities is to create an overlay that hides sessions which are being deleted and to clear the overlay on undo.

Thanks for understanding. The undo functionality was a hack that functioned best when the user couldn't delete multiple items easily. I explicitly created a new CoroutineScope so the operation wouldn't get cancelled when browsing away.